src/follows/follows.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable, UseFilters, UnauthorizedException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model, Types } from 'mongoose';
+import { BaseService } from '../common/base.service';
+import { Follows } from '../schema/follows/follows.interface';
+import { User } from '../schema/user/user.interface';
+import { UserService } from '../user/user.service';
+@Injectable()
+export class FollowsService extends BaseService<Follows> {
+
+    constructor(
+        @InjectModel('Follows') private readonly followsModel: Model<Follows>,
+        private readonly userService: UserService) {
+        super(followsModel);
+    }
+
+    async addFollow(user: User, followId: string) {
+        if (user.id === followId) {
+            throw new UnauthorizedException('Follow yourself');
+        }
+        const follow = await this.userService.findById(followId);
+        if (!follow) {
+            throw new UnauthorizedException('User does not exists');
+        }
+        const exists = await this.findOne({ uid: Types.ObjectId(user.id), followId: Types.ObjectId(followId) });
+        if (exists) {
+            throw new UnauthorizedException('User had followed');
```

Comment:
Was this [UnauthorizedException](https://github.com/nestjs/nest/blob/master/packages/common/exceptions/unauthorized.exception.ts) used by a mistake?

Reply:
yes,there just record error message.  need created new exception with error message and error code extends http exception.

Reply:
What do you mean by error code? Can you make an example?

Reply:
The error defined as follows:
export class UserIsNotExistsError extends HttpException {
    constructor() {
        super('user not exists', ErrorCode.USER_IS_NOT_EXISTS|| 10010);
    }
}

Reply:
OK,

BTW, you can use 3 back ticks to define code blocks.